osf_pigeon/pigeon.py

```diff
@@ -212,16 +217,46 @@ def get_with_retry(
 async def get_pages(url, page, result={}):
     url = f'{url}?page={page}'
     resp = get_with_retry(url, retry_on=(429,))
-    result[page] = resp.json()['data']
+
+    if 'contributors' in url:
```

Comment:
Instead of checking the url for a specific string, I'd just write a similar `get_pages_with_contributors` and `get_paginated_data_with_contributors` or get fancy and total generalize it like:

```python 
async def get_pages(url, page, result={}, parse_metadata=None):
    url = f'{url}?page={page}'
    resp = get_with_retry(url, retry_on=(429,))
    result[page] = resp.json()['data']

    if parse_metadata:
		result[page] = parse_metadata(resp.json())['data']

	return result


async def get_paginated_data(url, with_metadata=None):
    data = get_with_retry(url, retry_on=(429,)).json()

    tasks = []
    is_paginated = data.get('links', {}).get('next')

    if parse_metadata:
    	data = parse_metadata(data)

```

This will make it `get_and_write_json_to_temp` more reusable if we do something like:

```python
get_and_write_json_to_temp(
    f'{settings.OSF_API_URL}v2/registrations/{guid}/contributors/',
    temp_dir,
    'contributors.json',
    parse_metadata=get_contributors
)
```